src/layers/mapboxgl-layer.js

```diff
@@ -51,17 +60,98 @@ class MapboxLayerGL extends Layer {
   get visualChannels() {
     return {};
   }
+  datasetSelector = config => config.dataId;
+  gpuFilterSelector = (config, datasets) =>
+    (datasets[config.dataId] || {}).gpuFilter;
+  columnsSelector = config => pointColResolver(config.columns);
+
+  sourceSelector = createSelector(
+    this.datasetSelector,
+    this.columnsSelector,
+    (datasetId, columns) => `${datasetId}-${columns}`
+  );
+
+  filterSelector = createSelector(
+    this.gpuFilterSelector,
+    gpuFilter => gpuFilterToMapboxFilter(gpuFilter)
```

Comment:
based on gpuFilterSelector, gpuFilter may be null, you may want to make sure gpuFilterToMapboxFilter can handle null values

Reply:
done